packages/artillery-plugin-memory-inspector/test/index.spec.mjs

```diff
@@ -0,0 +1,205 @@
+import fs from 'fs';
+import { startTestServer } from './util.mjs';
+import { test, afterEach } from 'tap';
+import { $ } from 'zx';
+
+afterEach(async () => {
+  //cleanup output file after each test
+  fs.unlinkSync('./test/output.json');
+});
+
+test('cpu and memory metrics display in the aggregate report with the correct name', async (t) => {
+  //Arrange: Test Server and Plugin overrides
+  const { currentPort, childProcess, currentPid } = await startTestServer();
+  const override = JSON.stringify({
+    config: {
+      plugins: {
+        'memory-inspector': [{ pid: currentPid, name: 'express-example' }]
+      }
+    }
+  });
+
+  //Act: run the test and get report
+  const output =
+    await $`TEST_PORT=${currentPort} ../artillery/bin/run run ./test/fixtures/scenario.yml --overrides ${override} --output ./test/output.json`;
+  const report = JSON.parse(fs.readFileSync('./test/output.json'));
+
+  //Assert:
+  //plugin doesn't mess with existing before scenario handlers
+  t.ok(
+    output.stdout.includes(
+      'Hello from the Handler!',
+      'Issue with running existing Before Scenario Handler!'
+    )
+  );
+
+  //sanity check that it can reach server
+  t.ok(
+    report.aggregate.counters['http.codes.200'] > 0,
+    'Should have 200 status codes'
+  );
+
+  //assert that correct custom metrics are emitted
+  t.hasProp(
+    report.aggregate.summaries,
+    'express-example.cpu',
+    "Aggregate Summaries doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.summaries,
+    'express-example.memory',
+    "Aggregate Summaries doesn't have Memory metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    'express-example.cpu',
+    "Aggregate Histograms doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    'express-example.memory',
+    "Aggregate Histograms doesn't have Memory metric"
+  );
+
+  //Cleanup: server
+  childProcess.kill();
```

Comment:
Not sure how `tap` behaves in this regard, but test runners usually throw on early assertion failures. Can you confirm that's not the case here, and if any of the assertions above fail the test will still reach the cleanup code? 

If it won't, this is a recipe for memory leaks in tests. We can address it by ensuring the cleanup after all tests by other means (again, have to read into it, never worked with `tap`).

Reply:
At times like this, I wish testing frameworks would adopt test-scoped cleanups like so:

```js
test('test', body, cleanup)
```

I remember writing an extension for Jest that'd allow that. Should be a common practice since flat tests is a great thing to have.